event_model/__init__.py

```diff
@@ -1780,13 +1854,14 @@ class ComposeRunBundle:
     (i.e. compose_stream_resource).
     """
 
-    start_doc: dict
-    compose_descriptor: callable
-    compose_resource: callable
-    compose_stop: callable
-    compose_stream_resource: Optional[callable] = None
+    start_doc: RunStart
+    compose_descriptor: Callable
+    compose_resource: Callable
+    compose_stop: Callable
```

Comment:
```suggestion
    compose_descriptor: ComposeDescriptor
    compose_resource: ComposeResource
    compose_stop: ComposeStop
```
Also, could you move this class to just above ComposeRun please?